src/Crawler/RepositoryCrawler.cs

```diff
@@ -0,0 +1,208 @@
+                }
+
+                return existingRepository;
+            }
+
+            var (root, subsidiarys) = await AnalyseRepository(rootRepositoryUri, parent).ConfigureAwait(false);
+            if (root is not null && modelRepositories.TryAdd(rootRepositoryUri.Host, root))
+            {
+                var subsidiarySites = new HashSet<Repository>();
+                foreach (var subsidiary in subsidiarys)
+                {
+                    var subsidiaryRepository = await CrawlRepositories(subsidiary, root).ConfigureAwait(false);
+                    if (subsidiaryRepository != null)
+                    {
+                        subsidiarySites.Add(subsidiaryRepository);
+                    }
+                }
+
+                root.SubsidiarySites = subsidiarySites;
+            }
+
+            return root;
+        }
+
+        internal async Task<(Repository? Repository, IEnumerable<Uri> SubsidiaryRepositories)> AnalyseRepository(Uri repositoryUri, Repository? parent = null)
+        {
+            try
+            {
+                var ilisiteTask = ParseIlisite(repositoryUri);
+                var modelsTask = CrawlIlimodels(repositoryUri);
+                var metadatasTask = CrawlIlidata(repositoryUri);
+
+                var ilisite = await ilisiteTask.ConfigureAwait(false);
+                var models = await modelsTask.ConfigureAwait(false);
+                var metadatas = await metadatasTask.ConfigureAwait(false);
+
+                if (ilisite is null) return (null, Array.Empty<Uri>());
+
+                var repository = new Repository
+                {
+                    Uri = repositoryUri,
+                    Name = ilisite.Name ?? repositoryUri.Host,
+                    Title = ilisite.Title,
+                    ShortDescription = ilisite.shortDescription,
+                    Owner = ilisite.Owner,
+                    TechnicalContact = ilisite.technicalContact,
+                    Models = models,
+                    Catalogs = metadatas,
+                };
+
+                if (parent != null)
+                {
+                    repository.ParentSites.Add(parent);
+                }
+
+                foreach (var model in models)
+                {
+                    model.ModelRepository = repository;
+                }
+
+                var subsidiaryRepositories = ilisite.subsidiarySites?
+                        .Where(location => location?.value is not null)
+                        .Select(location => new Uri(location?.value ?? string.Empty))
+                        .ToArray() ?? Array.Empty<Uri>();
+
+                return (repository, subsidiaryRepositories);
+            }
+            catch (Exception ex) when (ex is HttpRequestException
+                                    || ex is InvalidOperationException)
+            {
+                Console.WriteLine($"Analysis of {repositoryUri} failed! ({ex.Message})");
+                Console.WriteLine(ex);
```

Comment:
REQ: `ILogger` für das Logging verwenden. Betrifft alle Stellen bei denen in die Konsole geloggt wird.